===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(swPipeBase_create:object)) at 6 (line 39, column 9) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/swoole-src/src/pipe/PipeBase.c
+++ b/table1/benchmarks/swoole-src/src/pipe/PipeBase.c
@@ -38,6 +38,7 @@ int swPipeBase_create(swPipe *p, int blocking)
     ret = pipe(object->pipes);
     if (ret < 0)
     {
+        free(object);
         swWarn("pipe create fail. Error: %s[%d]", strerror(errno), errno);
         return -1;
     }
